snow/engine/avalanche/transitive_test.go

```diff
@@ -145,6 +145,180 @@ func TestEngineAdd(t *testing.T) {
 	}
 }
 
+func TestEngineFulfillMissingDependencies(t *testing.T) {
+	config := DefaultConfig()
+
+	vals := validators.NewSet()
+	config.Validators = vals
+
+	vdr := ids.GenerateTestShortID()
+	if err := vals.AddWeight(vdr, 1); err != nil {
+		t.Fatal(err)
+	}
+
+	sender := &common.SenderTest{}
+	sender.T = t
+	config.Sender = sender
+
+	sender.Default(true)
+	sender.CantGetAcceptedFrontier = false
+
+	manager := vertex.NewTestManager(t)
+	config.Manager = manager
+
+	manager.Default(true)
+
+	manager.CantEdge = false
+
+	te := &Transitive{}
+	if err := te.Initialize(config); err != nil {
+		t.Fatal(err)
+	}
+
+	if te.Ctx.ChainID != ids.Empty {
+		t.Fatalf("Wrong chain ID")
+	}
+
+	trA := &conflicts.TestTransition{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Unknown,
+	}
+	txA := &conflicts.TestTx{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		TransitionV: trA,
+	}
+	trB := &conflicts.TestTransition{
+		IDV:           ids.GenerateTestID(),
+		StatusV:       choices.Processing,
+		DependenciesV: []conflicts.Transition{trA},
+	}
+	txB := &conflicts.TestTx{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		TransitionV: trB,
+	}
+
+	parentVtx := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		BytesV: []byte{1},
+		TxsV:   []conflicts.Tx{txA},
+	}
+	childVtx := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		ParentsV: []avalanche.Vertex{
+			parentVtx,
+		},
+		BytesV: []byte{2},
+		TxsV:   []conflicts.Tx{txB},
+	}
+
+	asked := new(bool)
+	reqID := new(uint32)
+	sender.GetF = func(inVdr ids.ShortID, requestID uint32, vtxID ids.ID) {
+		*reqID = requestID
+		if *asked {
+			t.Fatalf("Asked multiple times")
+		}
+		*asked = true
+		if vdr != inVdr {
+			t.Fatalf("Asking wrong validator for vertex")
+		}
+		if parentVtx.ID() != vtxID {
+			t.Fatalf("Asking for wrong vertex")
+		}
+	}
+
+	manager.ParseF = func(b []byte) (avalanche.Vertex, error) {
+		if !bytes.Equal(b, childVtx.Bytes()) {
+			t.Fatalf("Wrong bytes")
+		}
+		return childVtx, nil
+	}
+
+	if err := te.Put(vdr, 0, childVtx.ID(), childVtx.Bytes()); err != nil {
+		t.Fatal(err)
+	}
+
+	if !*asked {
+		t.Fatalf("Didn't ask for a missing vertex")
+	}
+	if len(te.vtxBlocked) != 1 {
+		t.Fatalf("Should have been blocking on 1 missing vertex dependency")
+	}
+	if len(te.txBlocked) != 1 {
+		t.Fatalf("Should have been blocking on 1 missing transition dependency")
+	}
+	if te.missingTransitions.Len() != 1 {
+		t.Fatalf("Should have had one missing transition")
+	}
+
+	manager.ParseF = func(b []byte) (avalanche.Vertex, error) {
+		if !bytes.Equal(b, parentVtx.Bytes()) {
+			t.Fatalf("Wrong bytes")
+		}
+		return parentVtx, nil
+	}
+
+	pushedParentVtx := new(bool)
```

Comment:
Is there a reason not to do `pushedParentVtx := false` instead of this? I think it's a little cleaner that way

Reply:
I think it's easier to avoid potential confusion about variable scoping rules and to use a pointer rather than a local variable.